src/main/java/org/mifos/processor/bulk/camel/routes/SplittingRoute.java

```diff
@@ -110,24 +137,40 @@ public void configure() throws Exception {
                 }
 
                 int subBatchCount = 1;
+                CsvSchema csvSchema = csvMapper.schemaFor(Transaction.class);
+                csvSchema = csvSchema.withHeader();
                 for (int i = 0; i < lines.size(); i += subBatchSize) {
+                    String subBatchId = UUID.randomUUID().toString();
                     String filename = UUID.randomUUID() + "_" + "sub-batch-" + subBatchCount + ".csv";
-                    FileWriter writer = new FileWriter(filename);
-                    writer.write(header);
+                    logger.info("SubBatch Id {}", subBatchId);
+
+                    List<Transaction> subBatchTransactions = new ArrayList<>();
                     for (int j = i; j < Math.min(i + subBatchSize, lines.size()); j++) {
-                        writer.write(lines.get(j) + System.lineSeparator());
+                        Transaction transaction = TransactionParser.parseLineToTransaction(lines.get(j));
+                        assert transaction != null;
+                        transaction.setBatchId(subBatchId); // Set the subBatchId for the transaction
+                        subBatchTransactions.add(transaction);
+                    }
+
+                    // Write the list of Transactions to the file
+                    File file = new File(filename);
+                    try (SequenceWriter writer = csvMapper.writer(csvSchema).writeValues(file)) {
+                        writer.writeAll(subBatchTransactions);
+                    } catch (IOException e) {
+                        logger.error("Failed to write sub-batch file: " + filename, e);
                     }
-                    writer.close();
```

Comment:
Why the writer close is removed?

Reply:
Since, I have used writer with try-with resource statement it automatically closes writer irrespectve of whether  writing occurs or an exception is thrown.